### PR TITLE
Persist closed tab state

### LINKED ISSUE #6161: Slice 4: Persist Opened/Closed Tabs
- Create a `persistSidebarConfig` in `src/sidebar/sidebarSlice.ts`
    - See `persistEditorConfig` in `src/pageEditor/slices/editorSlice.ts` for an example
    - Instead of a `blacklist`, we need `whitelist: ["closedTabs"]`
- Update `configureStore` in `src/sidebar/store.ts` to persist the sidebar reducer using the config
- If all tabs are closed, reopening the sidebar should display the mod launcher (set the `closeTabs[<mod-launcher-event-key>]` to `false`)

### PR DESCRIPTION
**Pull request by turbochef**
## What does this PR do?

- Closes #6161
- Persist the `closedTabs` state in the sidebar slice
- If you open an empty sidebar, the modlauncher will open

## Demo

- [Loom](https://www.loom.com/share/de0112dcaa514cbbbf78a5a7621dcb3a?sid=9f022714-a035-4841-9357-ae437aa0834f)

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/sidebar/sidebarSlice.ts [hunk 1]
```diff
@@ -324,17 +339,25 @@ const sidebarSlice = createSlice({
...
     activatePanel(state, { payload }: PayloadAction<ActivatePanelOptions>) {
       state.pendingActivePanel = null;
-
       const hasActive = state.forms.length > 0 || state.panels.length > 0;
 
       if (hasActive && !payload.force) {
         return;
       }
 
+      const visiblePanelCount = getVisiblePanelCount(state);
```
Comment (grahamlangford) [inline src/sidebar/sidebarSlice.ts:348]: We should include a comment explaining this logic.

**Comment by grahamlangford** [comment]
@turbochef See my notes about the tests and comments. Otherwise, I think this can be marked `Ready for review`.

File: src/sidebar/sidebarSlice.test.ts [hunk 2]
```diff
@@ -233,6 +234,127 @@ describe("sidebarSlice.addForm", () => {
...
+    );
+
+    expect(newState).toStrictEqual({
+      ...state,
+      activeKey: tabId,
+      closedTabs: { [tabId]: false },
+    });
+  });
+
+  it("setInitialPanels opens mod launcher if there are no visible panels", () => {
```
[the remark this answers is not in the document]
Comment (turbochef) [reply src/sidebar/sidebarSlice.test.ts:299]: Makes sense. I added checks for the the activekeys in each of these tests.

File: src/sidebar/sidebarSelectors.ts [hunk 3]
```diff
@@ -81,26 +82,5 @@ export const selectExtensionFromEventKey =
...
-    (panel) => !closedTabs[eventKeyForEntry(panel)]
-  );
-
-  return (
-    openPanels.length +
-    forms.length +
-    temporaryPanels.length +
-    (modActivationPanel ? 1 : 0)
-  );
-};
```
Comment (turbochef) [inline src/sidebar/sidebarSelectors.ts:106]: Abstracted this in `utils.ts` without any changes.

File: src/sidebar/sidebarSlice.ts [hunk 4]
```diff
@@ -205,12 +209,25 @@ const sidebarSlice = createSlice({
         modActivationPanel: ModActivationPanelEntry | null;
       }>
     ) {
+      // We need a visible count to prevent showing an empty sidebar. If there are no visible panels,
+      // we need to show the mod launcher. If there are visible panels, we need to hide the mod launcher.
```
Comment (grahamlangford) [inline src/sidebar/sidebarSlice.ts:213]: The second part of this comment isn't exactly true. If the modLauncher and another panel are both open, then both are shown. We could add functionality to hide the modLauncher if there are other visible panels like we do with activatePanel, but we would need @BrandonPxBx to weigh in on that.

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**